### DISCUSSION TITLE
key repeat script

### DISCUSSION THREAD

**Discussion by KyleBucksot** [Hammerspoon/hammerspoon#2757] [Q&A]
have been using autohotkey in the past and have this script that I cannot live without while playing world of warcraft. I suffered a hand injury and can't play without the following script attached

All it does is key repeat the letter r while holding it down. I use an addon in world of warcraft called gnomesequenceenhanced to put my abilities in to play. and the scritp uses the same button over and over until released to cast my abilities.

The old script is attached that worked in autohotkey on pc, i need a new script built in hammerspoon to do the exact same thing since switching to a m1 mac mini.

Hold down the r key, it repeats while world of warcraft window is active until i release it. 

Here is the script from autohotkey.

#ifWinActive World of Warcraft
{
$r::
   Loop  
   {
    if not GetKeyState("r", "P")
      break
     Send r
     Sleep 40
    }
return
}

**Comment by KyleBucksot** [comment]
And just want to go ahead and post this. it is not against the tos to use this script.
![image](https://user-images.githubusercontent.com/77978533/110974830-f6f54180-832c-11eb-8d8e-5f207f6dc9f0.png)

**Comment by latenitefilms** [comment]
You'll probably get better performance using: https://karabiner-elements.pqrs.org

**Comment by KyleBucksot** [comment]
Ok thanks I posted there before but never got a response. I’ll keep trying.

**Comment by KyleBucksot** [comment]
Here is the comment. I’m not sure why I’m doing wrong with it but seems like it’s not a function in karabiner anymore?

https://github.com/pqrs-org/Karabiner-Elements/issues/2598#issuecomment-768229177
